api/v1alpha1/backendtrafficpolicy_types.go

```diff
@@ -86,6 +86,11 @@ type BackendTrafficPolicySpec struct {
 	//
 	// +optional
 	Timeout *Timeout `json:"timeout,omitempty"`
+
+	// The compression config for the http streams.
+	//
+	// +optional
+	Compression []*Compression `json:"compression,omitempty"`
```

Comment:
why this is a list ?

Reply:
to support enable multiple compression algorithm

Reply:
is that a valid use case ?

Reply:
rethinking, feels like a valid use case to enable both gzip and brotli and decide based on client capability.
for that case, should the config look more like
```
compression:
  gzip: {}
  brotli: {}
```

the type is no longer needed since we aren't picking 1
a list also works, but extra validation needs to be added to ensure a specific type is not enabled multiple times

Reply:
yes, at least from the Envoy side https://github.com/envoyproxy/gateway/pull/2474#discussion_r1465707302

There is also option `choose_first` https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/filters/http/compressor/v3/compressor.proto for the case of multiple compressors.

Reply:
cool, in the future, we could consider adding a `priority` field to orchestrate order

Reply:
> rethinking, feels like a valid use case to enable both gzip and brotli and decide based on client capability. for that case, should the config look more like
> 
> ```
> compression:
>   gzip: {}
>   brotli: {}
> ```
> 
> the type is no longer needed since we aren't picking 1 a list also works, but extra validation needs to be added to ensure a specific type is not enabled multiple times

@arkodg sorry, just found I missed this comment.

I use the list since I'm afraid there are configs may different for the different compressors. Like they may use the different `content_length`. If we put them into one `compression` message, then we will block this case.

Reply:
can you share a use case where we'd define `gzip` compression twice ?

Reply:
no, I don't think we are going to define the same compression twice. I think it is all about multiple different compression.

Reply:
if so, prefer 
```yaml
compression:
  gzip: {}
  brotli: {}
```

Reply:
in that case, imo we shouldnt use a list (which makes validation hard), instead use something like
```
compression:
  gzip: {}
  brotli: {}
 ```

Reply:
emm...sorry, I found a case for multiple same compressor. 

```
    // There could be many compressors registered for the same content encoding, e.g. consider a
    // case when there are two gzip filters using different compression levels for different content
    // sizes. In such case we ignore duplicates (or different filters for the same encoding)
    // registered last.
```
https://github.com/envoyproxy/envoy/blob/45ab9cfa30c7631fe325847dd291e2dcc80c1894/source/extensions/filters/http/compressor/compressor_filter.cc#L389-L392

Reply:
And we probably need the order of multiple compressors also.